packages/ethereum-contracts/contracts/agreements/gdav1/GeneralDistributionAgreementV1.sol

```diff
@@ -305,49 +311,102 @@ contract GeneralDistributionAgreementV1 is AgreementBase, TokenMonad, IGeneralDi
         pool.claimAll(memberAddress);
     }
 
-    // @note setPoolConnection function naming
-    function connectPool(ISuperfluidPool pool, bool doConnect, bytes calldata ctx)
-        public
-        returns (bytes memory newCtx)
+    /// @inheritdoc IGeneralDistributionAgreementV1
+    function connectPool(ISuperfluidPool pool, bytes calldata ctx) external override returns (bytes memory newCtx) {
+        newCtx = ctx;
+        _setPoolConnection(pool, address(0), true /* doConnect */, ctx);
```

Comment:
Consider to piggyback a small change `_setPoolConnection` to `_setPoolConnectionFor`, the code might read more fluidly gramatically.